### PR TITLE
[11.0][account_financial_report] filters, default dates, currency, partners

### PR DESCRIPTION
**Pull request by i-vyshnevska**
port a bunch of fixes and improvements from v10
[add report to res partner, default values](https://github.com/OCA/account-financial-reporting/pull/508)
[filter for partner](https://github.com/OCA/account-financial-reporting/pull/502)
[cherry-pick commit -  fix currency onclick](https://github.com/OCA/account-financial-reporting/pull/503)

### CODE REVIEW

**Comment by MiquelRForgeFlow** [comment]
@i-vyshnevska conflicts

**Comment by i-vyshnevska** [comment]
where closed accidentally :)

File: account_financial_report/wizard/general_ledger_wizard.py [hunk 1]
```diff
@@ -80,9 +84,39 @@ class GeneralLedgerReportWizard(models.TransientModel):
...
              'account currency is not setup through chart of accounts '
-             'will display initial and final balance in that currency.'
+             'will display initial and final balance in that currency.',
+        default=lambda self: self._default_foreign_currency(),
     )
 
+    def _init_date_from(self):
+        """set start date to begin of current year if fiscal year running"""
+        today = fields.Date.context_today(self)
+        cur_month = int(fields.Date.from_string(today).strftime('%m'))
```
Comment (TDu) [inline account_financial_report/wizard/general_ledger_wizard.py]:
```suggestion
        cur_month = today.month
```

File: account_financial_report/wizard/general_ledger_wizard.py [hunk 2]
```diff
@@ -80,9 +84,39 @@ class GeneralLedgerReportWizard(models.TransientModel):
...
+        cur_day = int(fields.Date.from_string(today).strftime('%d'))
```
Comment (TDu) [inline account_financial_report/wizard/general_ledger_wizard.py]:
```suggestion
        cur_day = today.day
```

**Review by TDu — COMMENTED** [review]
A couple of improvements maybe

**Review by sebalix — APPROVED** [review]
Rebase for fixup commit needed

File: account_financial_report/wizard/open_items_wizard.py [hunk 3]
```diff
@@ -44,14 +44,33 @@ class OpenItemsReportWizard(models.TransientModel):
...
         string='Show foreign currency',
         help='Display foreign currency for move lines, unless '
              'account currency is not setup through chart of accounts '
-             'will display initial and final balance in that currency.'
+             'will display initial and final balance in that currency.',
+        default=lambda self: self._default_foreign_currency(),
     )
 
+    def _default_foreign_currency(self):
+        if self.env.user.has_group('base.group_multi_currency'):
```
Comment (grindtildeath) [inline account_financial_report/wizard/open_items_wizard.py]:
```suggestion
        return self.env.user.has_group('base.group_multi_currency')
```

File: account_financial_report/tests/test_general_ledger.py [hunk 4]
```diff
@@ -493,3 +494,57 @@ def test_04_unaffected_account_balance_2_years(self):
...
+    def test_validate_date(self):
+        company_id = self.env.ref('base.main_company')
+        company_id.write({
+            'fiscalyear_last_day': 31,
+            'fiscalyear_last_month': 12,
+        })
+        cr = self.registry.cursor()
+        user = self.env.ref('base.user_root').with_context({
+            'company_id': company_id.id})
+        env = Environment(cr, user.id, {})
```
Comment (grindtildeath) [inline account_financial_report/tests/test_general_ledger.py]: Any good reason to declare this env using the same cursor ?

File: account_financial_report/tests/test_general_ledger.py [hunk 5]
```diff
@@ -493,3 +494,57 @@ def test_04_unaffected_account_balance_2_years(self):
...
+        wizard = env["general.ledger.report.wizard"]
+        self.assertEqual(wizard._init_date_from(),
+                         time.strftime('%Y') + '-01-01')
+
+    def test_validate_date_range(self):
+        type = self.env['date.range.type'].create({
```
Comment (grindtildeath) [inline account_financial_report/tests/test_general_ledger.py]: Please do not obfuscate builtins

**Review by grindtildeath — COMMENTED** [review]
Some nitpicking

File: account_financial_report/tests/test_general_ledger.py (hunk 4 above)
Comment (i-vyshnevska) [reply account_financial_report/tests/test_general_ledger.py]: leftover from 10 version, thanks,
 updated

File: account_financial_report/wizard/general_ledger_wizard.py [hunk 6]
```diff
@@ -80,9 +84,39 @@ class GeneralLedgerReportWizard(models.TransientModel):
...
+        cur_day = fields.Date.from_string(today).day
+        last_fsc_month = self.env.user.company_id.fiscalyear_last_month
+        last_fsc_day = self.env.user.company_id.fiscalyear_last_day
+
+        if cur_month < last_fsc_month \
+                or cur_month == last_fsc_month and cur_day <= last_fsc_day:
+            return time.strftime('%Y-01-01')
+
+    def _default_foreign_currency(self):
+        if self.env.user.has_group('base.group_multi_currency'):
```
Comment (grindtildeath) [inline account_financial_report/wizard/general_ledger_wizard.py]:
```suggestion
        return self.env.user.has_group('base.group_multi_currency')
```
Sorry I missed this one on the last review.

Comment (i-vyshnevska) [reply account_financial_report/wizard/general_ledger_wizard.py]: updated

**Review by TDu — APPROVED** [review]

**Review by TDu — COMMENTED** [review]
Was a little to fast still a little fix needed for the linter

**Comment by JordiBForgeFlow** [comment]
This PR introduces a change by which anyone can access to a partner and run the general ledger and open items ledger for the whole company. We need to revert https://github.com/OCA/account-financial-reporting/blob/11.0/account_financial_report/wizard/general_ledger_wizard_view.xml#L89 ASAP

**Comment by i-vyshnevska** [comment]
maybe group for action will solve problem?

**Comment by pedrobaeza** [comment]
Yes, I think a group in the action will do the trick.

**Comment by i-vyshnevska** [comment]
done in https://github.com/OCA/account-financial-reporting/pull/577

**Comment by pedrobaeza** [comment]
@jbeficent I have fast-tracked the PRs with the fix. Please check that everything is OK.

**Pull request merged**